### ISSUE TITLE
Adding an association updates the parent model with all of its existing values

### ISSUE THREAD

**Issue by mmiller42** [balderdashy/sails#3103]
Let's say I have a 1:many relationship, i.e. a given `User` has *n* `Picture`s.

The issue is, I've run into a number of validation issues that I think have to do with the underlying implementation:

(1) The `Picture` model might be set up as so:

```js
module.exports = {
	attributes: {
		url: {
			type: 'string',
			required: true
		},
		owner: {
			model: 'user',
			required: true
		}
	}
};
```

For some reason, the `owner` attribute is not being populated before being saved, so when requesting `POST /users/42/pictures` with `{ "url": "http://whatever/test.jpg" }` returns a validation error. However, if I lift the required restriction on `owner`, it *does* populate the field eventually (if I query it in Postgres, `user` is ultimately set to `42`). I don't want to remove this restriction because if I am creating a picture via `POST /pictures`, the `owner` should be required.

(2) I know Waterline is database-agnostic, so in my case (using Postgres), I'd imagine that the `Picture` table would have a `user` column with its associated user ID, whereas if I were using Mongo, perhaps the `User` table would have an array of `Picture` IDs (or objects, depending on implementation). Therefore, when adding a picture, even though I'm using Postgres, I'd imagine that the user might be "updated."

Unfortunately, the user is always being updated when adding a picture associated with the user, and more importantly, *all* of its attributes are passed into the `beforeUpdate` and `beforeCreate` hooks, even though they are not changed. So, for example, in a `beforeUpdate` hook, the value of the `password` field is hashed with bcrypt, if it is present in the `valuesToUpdate` argument; upon adding a picture to the user, the hash is again hashed, even though it shouldn't be in `valuesToUpdate`.

**Comment by sailsbot** [comment]
Thanks for posting, @mmiller42.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
